### DISCUSSION TITLE
Allow the `.server` and `.client` to work outside `appDirectory`

### DISCUSSION THREAD

**Discussion by chrisfitz** [remix-run/remix#4292] [Proposals]
As it stands today, the `.server` and `.client` convention _only_ works within the `appDirectory` (see [here](https://github.com/remix-run/remix/blob/eafad35a480ed6bb61fdbe703680893e3b4cb70b/packages/remix-dev/compiler/plugins/emptyModulesPlugin.ts#L19-L26)), however this is quite limiting if you're working within a monorepo-style repository or have some code lives outside the `appDirectory`.

### Consider the Following
```
app/
build/
public/
node_modules/
libs/
├─ example-library/
│  ├─ example-file.server.ts
...
```

As it stands, if you attempt to name a file inside the example library with `.server` or `.client`, it will have no effect. It would be great if there was an option within `AppConfig` to allow you to specify directories for this to work in rather than hardcoding just the it to `appDirectory`.

Is this something that would be accepted as a PR?

**Comment by jrestall** [comment]
Would also love to see this changed, currently also using a monorepo and having to patch Remix's emptymodulesplugin code to ensure server code in libs is removed from the browser bundle as discussed at https://github.com/remix-run/blues-stack/pull/65#issuecomment-1243188315

I'm sure there is a better way to exclude node_modules from this convention without excluding everything outside the app folder.
